tensorhive/models/Group.py

```diff
@@ -34,21 +35,45 @@ def users(self):
         return self._users
 
     def add_user(self, user: User):
+        if user in self.users:
+            raise InvalidRequestException('User {user} is already a member of group {group}!'
+                                          .format(user=user, group=self))
         self.users.append(user)
         self.save()
 
     def remove_user(self, user: User):
+        if user not in self.users:
+            raise InvalidRequestException('User {user} is not a member of group {group}!'
+                                          .format(user=user, group=self))
+
         self.users.remove(user)
         self.save()
 
-    @property
+    @hybrid_property
     def as_dict(self):
-        return {
+        """
+        Serializes current instance into dict.
+        :return: Dictionary representing current instance.
+        """
+        return self._as_dict(True)
+
+    @hybrid_property
+    def as_dict_shallow(self):
+        """
+        Serializes current instance into dict. Will not include group's users (to prevent recurrence).
+        :return: Dictionary representing current instance (without users).
+        """
+        return self._as_dict(False)
+
+    def _as_dict(self, include_users):
+        group = {
             'id': self.id,
             'name': self.name,
-            'createdAt': self.created_at.isoformat(),
-            'users': self.users
+            'createdAt': self.created_at.isoformat()
```

Comment:
I'd prefer to stick to the isoformat for this field, as this field was already there before my changes, meaning that some entries in the DB already follow the format of `.isoformat()`. Changing that would lead to having one column with two different formats, which could cause some headaches.

Reply:
The problem is that we have two formats in use already:
* `isoformat` in:
** User model - `created_at`

* `DateUtils` format in:
** Reservation model - `created_at`, `start`, `end`
** Task model - `spawn_at`, `terminate_at`

(the models added by me all use DateUtils helper).
Since the isoformat is only used when displaying when user was created in the admin panel, I guess we could switch to DateUtils entirely (verified that it works both for DateUtils and isoformat). The newest commit addresses that.